### ISSUE TITLE
Creation of RFID Logs

### ISSUE THREAD

**Issue by roquegv** [phamos-eu/zebra#2]
The current approach to create new RFID Logs has the following limitations:
1. The RFID Reader sends sends hundreds and even thousands of reading every hour, but we don't need to store every reading but only some of them. For example, we may only need to create a RFID Log once every 60 minutes. As far as we tried, this is not something we can configure in the Reader's control panel. So, we need to filter the readings in our endpoint.
2. Every time the reader sends data, it does in bulk: many readings for the same tag. We don't need to create a RFID Log of each of then, but only the last one.
3. There could be cases in which the RFID tag is not "visible" by the reader. The tag should be considered to be in a different location only if it seem in another location (from another reader).

Tasks:

- [x] Make a filter in the endpoint to just create RFID Log every X minutes.
- [ ] This X minutes should configurable, for example in a Zebra Setting single doctype.
- [x] Make a filter to take just the last reading from a set of readings that the reader sends each time.
- [x] Change tag/equipment's location only when it is read by another reader

**Comment by wojosc** [comment]
Just to make sure: on your point 1. 60 minutes would be way to far apart from readings. Max would be some seconds as we need to concider, that passing through a door with equipment will only be seconds.

**Comment by roquegv** [comment]
> Just to make sure: on your point 1. 60 minutes would be way to far apart from readings. Max would be some seconds as we need to concider, that passing through a door with equipment will only be seconds.

Yes, maybe good idea to set this in a Zebra Settings page (single doctype).

**Comment by wojosc** [comment]
that sounds like a good idea. Maybe it makes sense to have an extra docType "Reader Configuration" that can be linked into the Reader. Doing this we can have a set of different configuraitons in the system that can be applied to several different readers. This would also implement the solution of being able to manager differenz configuraitons on the same site.

E.g. you might have an office building in which we want the readers to be in the operating mode "simple" reading every second on sight. The same site might have a storage building though in which a different configuration applies where they will wan't to use the operating mode "inventory".

Does that make sense?

**Comment by roquegv** [comment]
Sounds good. Not sure though if we can control, via zebra's API, such settings like "operating mode". If we have full control over it, then this approach would works.

**Comment by roquegv** [comment]
Apparently yes, we have control over such settings:

![Captura de pantalla de 2023-04-11 09-02-29](https://user-images.githubusercontent.com/6966715/231170713-79a0c53c-36f7-4d71-be6e-8ab969b8e017.png)

Ref: https://zebradevs.github.io/rfid-ziotc-docs/api_ref/local_rest/index.html#local-rest-apis

**Comment by roquegv** [comment]
- Make a filter in the endpoint to just create RFID Log every X minutes
- Make a filter to take just the last reading from a set of readings that the reader sends each time

![Captura de pantalla de 2023-04-13 14-12-14](https://user-images.githubusercontent.com/6966715/231847401-904d5107-4e52-4693-bfc6-6d9c4e5bfb20.png)
(look at the creation time, logs are created one every 60 seconds (this is configurable).. in this case only 5 tags are read)

This basically means that only X records would be created every Y seconds (where X = amount of RFID tags and Y = interval in seconds).

- Change tag/equipment's location only when it is read by another reader

![Captura de pantalla de 2023-04-13 14-14-16](https://user-images.githubusercontent.com/6966715/231847825-3f2bc42b-7685-4c66-89e5-9d77ebac193b.png)
(Equipment is updated only if current location is different than previous location.. for testing purposes, this is achieved by changing RFID Reader's location)

**Comment by roquegv** [comment]
QUESTIONS: 
1. In the Equipment doctype there is a Item Code field, which is mandatory. What should be the value for this? 
2. The current code we have does not create an Item when an Equipment is created. Should we do that? If so, what should be the values?

**Comment by wojosc** [comment]
> QUESTIONS:
> 
>     1. In the Equipment doctype there is a Item Code field, which is mandatory. What should be the value for this?
> 
>     2. The current code we have does not create an Item when an Equipment is created. Should we do that? If so, what should be the values?

1. The mandetory field item_code was used to create new Equipment based on the same Item of ERPNext. This necessicty in the mean time has been deprecated. There is a system in place on which you can see how it is setup so far. I will share some credentials to that with you. The field will not need to be mendatory anymore or might even cease to exist.

2. What do you mean here? The customer is mainly creating Equipment which where so far based on Items (see 1. ). They are not created automatically in any way. The way it will work is: a) create new Equipment b) set an RFID-Tag in the respective field c) print an RFID-Tag and stick it to the actual physical equipment. From this moment on the Equipment can be used with the RFID-System.
The idea used to be to have the Item created before it is linked with an Equipment. It worked kind of like a template. E.G. you have a red a blue and a green coffee cup (Equipment) that are all based on a coffee cup (Item).

**Comment by wojosc** [comment]
> * This X minutes should configurable, for example in a Zebra Setting single doctype.

@roquegv I guess this could be set on two levels. Level 1 could be an overall settings page for the app and on Level 2 being set individually on each reader as we might want to have different settings for different locations. In this case primarily having it on a overall settings page will do fine.
